helm_helpers.go

```diff
@@ -342,28 +342,46 @@ func addHelmRepos(repos map[string]string) (bool, string) {
 // If serviceAccount is not provided (empty string), the defaultServiceAccount is used.
 // If no defaultServiceAccount is provided, A service account is created and Tiller is deployed with the new service account
 // If no namespace is provided, Tiller is deployed to kube-system
-func deployTiller(namespace string, serviceAccount string, defaultServiceAccount string, role string) (bool, string) {
+func deployTiller(namespace string, serviceAccount string, defaultServiceAccount string, role string, roleTemplateFile string) (bool, string) {
 	log.Println("INFO: deploying Tiller in namespace [ " + namespace + " ].")
 	sa := ""
 	if serviceAccount != "" {
 		if ok, err := validateServiceAccount(serviceAccount, namespace); !ok {
 			if strings.Contains(err, "NotFound") || strings.Contains(err, "not found") {
 
 				log.Println("INFO: service account [ " + serviceAccount + " ] does not exist in namespace [ " + namespace + " ] .. attempting to create it ... ")
-				if _, rbacErr := createRBAC(serviceAccount, namespace, role); rbacErr != "" {
+				if _, rbacErr := createRBAC(serviceAccount, namespace, role, roleTemplateFile); rbacErr != "" {
 					return false, rbacErr
 				}
 			} else {
 				return false, "ERROR: while validating/creating service account [ " + serviceAccount + " ] in namespace [" + namespace + "]: " + err
 			}
 		}
 		sa = "--service-account " + serviceAccount
+	} else if serviceAccount == "" && roleTemplateFile != "" {
+		roleName := "helmsman-tiller"
+		defaultServiceAccountName := "helmsman"
```

Comment:
Thank you @mkubaczyk for this PR .. I have wanted to add this for a long time but didn't get around to it :)
I have a small comment on the `defaultServiceAccountName` here .. I am wondering why you reset here even though it might hold a value which is passed to the deployTiller function. 

I would suggest to check if defaultServiceAccountName has a value and use it, otherwise, set its value to helmsman. This way, you can also get rid of the third else if below in line 379.

Reply:
Hi, thanks for an idea. I did what you've asked for. Code in fact became a lot easier in this function. :-)